src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -523,7 +566,7 @@ private static NullableAnnotation MergeNullableAnnotation(TypeSymbol type, Nulla
         /// <summary>
         /// The list of custom modifiers, if any, associated with the <see cref="TypeSymbol"/>.
         /// </summary>
-        public ImmutableArray<CustomModifier> CustomModifiers => _extensions.CustomModifiers;
+        public ImmutableArray<CustomModifier> CustomModifiers => _extensions is null ? default : _extensions.CustomModifiers;
```

Comment:
>_extensions is null [](start = 65, length = 19)

Are there callers of `type.CustomModifiers` that are not checking `type.IsNull`?

Reply:
The method I added (`AsSpeakable`) would run into this.

Reply:
I don't mind making that change, but I'm curious what is the benefit.
When we create a TSWA, we can pass `default` custom modifiers. Why throw instead of returning `default` back when asked for the customer modifiers?

Reply:
`CustomModifiers` is never `default` for an initialized `TypeSymbolWithAnnotations`.

We could have these properties (and other `TypeSymbolWithAnnotations` members) handle uninitialized (`IsNull`) instances but that will add some complexity to these members and might hide bugs where the caller should have handled `IsNull` explicitly.

---
In reply to: [242803704](https://github.com/dotnet/roslyn/pull/31813#discussion_r242803704) [](ancestors = 242803704)